src/ray/protobuf/core_worker.proto

```diff
@@ -71,6 +71,15 @@ message ReturnObject {
   int64 size = 6;
 }
 
+message StealWorkRequest {
+  int64 max_tasks_to_steal = 1;
+}
+
+message StealWorkReply {
+  int64 number_of_tasks_stolen = 1;
+  repeated TaskSpec tasks_stolen = 2;
```

Comment:
Doesn't the owner already know which specs are on the worker? No need to send the full spec message back, can just send their indices.

Reply:
Hmm yeah, good point. It's probably best to send back the task ID instead of the indices since the indices could get updated between when the worker replies and when the owner receives the reply, though.

Reply:
Currently, I don't think the owner keeps a {`TaskId` -> `TaskSpecification`} mapping for the tasks that are in-flight to a worker. In fact, right after calling `PushNormalTask` to push a task to a worker, `OnWorkerIdle` pops the task's TaskSpec from the relevant owner's queue. So once we push a task, the owner will not see the task's TaskSpec until the worker responds to the owner and activates the `PushNormalTask` callback (which captures the TaskSpec).

For this reason, when we perform work stealing, the only way for the owner to access the the stolen tasks' TaskSpecs is for the victim to send them back along with the `StealWork` rpc reply. 

The worker will eventually also respond to each of the stolen tasks' original `PushNormalTask` RPCs, so an alternative approach would be to let those `PushNormalTask` callbacks take care of pushing the stolen tasks to the thief. However, this approach would be less efficient, because the replies would only start coming back after the worker has finished executing the non-stolen tasks that were pushed to the worker's queue before the stolen ones.